src/pe/load_config.rs

```diff
@@ -0,0 +1,1439 @@
+    /// Virtual address of the Guard CF function table.
+    pub guard_cf_function_table: u32,
+    /// Count of Guard CF functions.
+    pub guard_cf_function_count: u32,
+    /// Flags related to Control Flow Guard (CFG).
+    pub guard_flags: u32,
+    /// Code integrity configuration.
+    pub code_integrity: LoadConfigCodeIntegrity,
+    /// Virtual address of the Guard address-taken IAT entry table.
+    pub guard_address_taken_iat_entry_table: u32,
+    /// Count of Guard address-taken IAT entries.
+    pub guard_address_taken_iat_entry_count: u32,
+    /// Virtual address of the Guard long jump target table.
+    pub guard_long_jump_target_table: u32,
+    /// Count of Guard long jump targets.
+    pub guard_long_jump_target_count: u32,
+    /// Virtual address of the dynamic value relocation table.
+    pub dynamic_value_reloc_table: u32,
+    /// Virtual address of the CHPE metadata.
+    pub chpe_metadata_pointer: u32,
+    /// Virtual address of the Guard RF failure routine.
+    pub guard_rf_failure_routine: u32,
+    /// Virtual address of the Guard RF failure routine function pointer.
+    pub guard_rf_failure_routine_function_pointer: u32,
+    /// Offset of the dynamic value relocation table.
+    pub dynamic_value_reloc_table_offset: u32,
+    /// Section index of the dynamic value relocation table.
+    pub dynamic_value_reloc_table_section: u16,
+    /// Reserved field.
+    pub reserved2: u16,
+    /// Virtual address of the Guard RF verify stack pointer function pointer.
+    pub guard_rf_verify_stack_pointer_function_pointer: u32,
+    /// Offset to the hot patch table.
+    pub hot_patch_table_offset: u32,
+    /// Reserved field.
+    pub reserved3: u32,
+    /// Virtual address of the enclave configuration pointer.
+    pub enclave_configuration_pointer: u32,
+    /// Virtual address of the volatile metadata pointer.
+    pub volatile_metadata_pointer: u32,
+    /// Virtual address of the Guard EH continuation table.
+    pub guard_eh_continuation_table: u32,
+    /// Count of Guard EH continuations.
+    pub guard_eh_continuation_count: u32,
+    /// Virtual address of the Guard XFG check function pointer.
+    pub guard_xfg_check_function_pointer: u32,
+    /// Virtual address of the Guard XFG dispatch function pointer.
+    pub guard_xfg_dispatch_function_pointer: u32,
+    /// Virtual address of the Guard XFG table dispatch function pointer.
+    pub guard_xfg_table_dispatch_function_pointer: u32,
+    /// Virtual address of the CASTGuard OS-determined failure mode handler.
+    pub cast_guard_os_determined_failure_mode: u32,
+    /// Virtual address of the Guard memcpy function pointer.
+    pub guard_memcpy_function_pointer: u32,
+}
+
+/// Macro that checks if the given structure has a field available
+/// based on a size check.
+macro_rules! have_field {
+    ($bytes:expr, $size:expr, $offset:expr, $ty:ty) => {
+        ($size as usize >= $offset + ::core::mem::size_of::<$ty>())
```

Comment:
i think you don't need to do this check; a raw `$bytes.pread::<$ty>($offset).ok()` should be the same thing as checking the offset + size (this is what pread does and why it can error, i mean technically it calls `SizeWith` for dynamic parsing contexts, but numeric values have none)